flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -897,7 +897,7 @@ class TaskManager(
         config.timeout,
         libCache,
         fileCache,
-        runtimeInfo)
+        new TaskRuntimeInfo(hostname, taskManagerConfig, tdd.getAttemptNumber))
```

Comment:
generally we try to keep one PR for one issue, exceptions should only be done for closely related issues.

why did you decide to add these issues into this PR? ( i have a hard time understanding it, since the commits barely touch the same files.